### PR TITLE
Add 'image-rendering' support and some cleanup

### PR DESCRIPTION
**Pull request by ghost**
---

- [x] `./mach build-geckolib` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] These changes have been tested manually

screenshot

r? @Manishearth 

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13636)

### CODE REVIEW

File: components/style/properties/gecko.mako.rs [hunk 1]
```diff
@@ -1445,6 +1437,17 @@ fn static_assert() {
 </%self:impl_trait>
 
 
+<%self:impl_trait style_struct_name="InheritedBox"
+                  skip_longhands="image-rendering">
+
+    <% render_keyword = Keyword("image-rendering",
+                                "auto optimizeQuality optimizeSpeed crispEdges") %>
+
+    ${impl_keyword('image_rendering', 'mImageRendering', render_keyword, need_clone=False, style='camel')}
```
Comment (Manishearth) [inline components/style/properties/gecko.mako.rs]: I would prefer if we just used a snake case enum for this struct, that way further helpers don't have to specialize on the style

File: components/style/properties/longhand/inherited_box.mako.rs [hunk 2]
```diff
@@ -50,15 +50,25 @@ ${helpers.single_keyword("color-adjust",
         pub enum T {
             Auto,
             CrispEdges,
-            Pixelated,
+            % if product == "gecko":
+                OptimizeQuality,
+                OptimizeSpeed,
+            % else:
+                Pixelated,
```
Comment (Manishearth) [inline components/style/properties/longhand/inherited_box.mako.rs]: Link to https://bugzilla.mozilla.org/show_bug.cgi?id=856337

**Comment by Manishearth** [comment]
I can't actually see the image rendering quality in the screenshots. I assume they work locally and there's a visible difference between before and after the patch?

**Comment by Wafflespeanut** [comment]
Addressed the changes :)

**Comment by Wafflespeanut** [comment]
Thanks! :)

**Comment by Manishearth** [comment]
@bors-servo retry
 - https://github.com/servo/servo/issues/13643

**Pull request merged**